pimd/pim_mroute.c

```diff
@@ -615,17 +615,9 @@ static int pim_mroute_msg(struct pim_instance *pim, const char *buf,
 			return 0;
 
 		connected_src = pim_if_connected_to_source(ifp, ip_hdr->ip_src);
-
```

Comment:
The RFC quoted says that the ip_src address of 0.0.0.0 must be accepted.  This change accepts *any* non-connected address that may be sent.  Which is not correct..   I would say leave line 619 in but change the if statement to `if (!connected_src && ip_hdr->ip_src != INADDR_ANY)